packages/ubuntu_wsl_setup/integration_test/ubuntu_wsl_setup_test.dart

```diff
@@ -0,0 +1,379 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:ini/ini.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:path/path.dart' as p;
+import 'package:subiquity_client/subiquity_client.dart';
+import 'package:subiquity_client/subiquity_server.dart';
+import 'package:ubuntu_wizard/widgets.dart';
+import 'package:ubuntu_wsl_setup/l10n.dart';
+import 'package:ubuntu_wsl_setup/main.dart' as app;
+import 'package:ubuntu_wsl_setup/pages.dart';
+import 'package:ubuntu_wsl_setup/routes.dart';
+
+import '../test/test_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  void expectPage(
+    WidgetTester tester,
+    Type page,
+    String Function(AppLocalizations lang) title,
+  ) {
+    LangTester.type = page;
+    expect(find.byType(page), findsOneWidget);
+    expect(find.widgetWithText(AppBar, title(tester.lang)), findsOneWidget);
+  }
+
+  Future<void> testSelectYourLanguage(
+    WidgetTester tester, {
+    String? language,
+  }) async {
+    expectPage(tester, SelectLanguagePage, (lang) => lang.selectLanguageTitle);
+
+    if (language != null) {
+      await tester.tap(find.widgetWithText(ListTile, language));
+      await tester.pump();
+    }
+    await tester.pumpAndSettle();
+
+    await tester.tapContinue();
+  }
+
+  Future<void> testProfileSetup(
+    WidgetTester tester,
+    IdentityData? profile, {
```

Comment:
I didn’t see any code path where you call it with null, is it intended to be nullable?
Same for the 3 other fields

Reply:
The idea was that one could surf through pages without having to pass anything in test cases where that page is irrelevant.